### PR TITLE
Add metadata harmonization options to `load_spectra` function

### LINKED ISSUE #442: Add metadata harmonization options to `load_spectra` function

### LINKED ISSUE #444: Update CI from Node12 actions

### PR DESCRIPTION
**Pull request by hechth**
Fixes #442
Fixes #444

### CODE REVIEW

File: matchms/importing/load_spectra.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import pickle
-from typing import Any, Generator, Optional
+from typing import Generator, Optional
 from matchms.importing import (load_from_json, load_from_mgf, load_from_msp,
                                load_from_mzml, load_from_mzxml, load_from_usi)
 from matchms.typing import SpectrumType
 
 
-def load_spectra(file: str, ftype: Optional[str] = None) -> Generator[SpectrumType, None, None]:
+def load_spectra(file: str, metadata_harmonization: bool = True, ftype: Optional[str] = None) -> Generator[SpectrumType, None, None]:
```
Comment (niekdejonge) [inline matchms/importing/load_spectra.py:9]: Makes sense to add this indeed!

File: matchms/importing/load_spectra.py [hunk 2]
```diff
@@ -29,26 +29,24 @@ def load_spectra(file: str, ftype: Optional[str] = None) -> Generator[SpectrumTy
...
-        assert isinstance(spectra[0], SpectrumType), "Expected list of spectra"
-        return spectra
-    assert False, f"File extension of file: {file} is not recognized"
+        return load_from_pickle(file, metadata_harmonization)
+        
+    raise TypeError(f"File extension of file: {file} is not recognized")
 
 
-def load_from_pickle(filename: str) -> Any:
+def load_from_pickle(filename: str, metadata_harmonization: bool) -> SpectrumType:
```
Comment (niekdejonge) [inline matchms/importing/load_spectra.py:49]: Good to add the check here. Maybe change the name to load spectra_from_pickle?

**Review by niekdejonge — APPROVED** [review]

**Review by florian-huber — APPROVED** [review]
Nice work, thanks!
@hechth feel free to merge when you are ready.

**Comment by florian-huber** [comment]
@hechth I am preparing a new release, so I will merge this. Hope that's OK.
Thanks for the great fixes!

**Pull request merged**